### PR TITLE
Reducer kernel in pure DPC++ 

### PR DESCRIPTION
**Pull request by KulikovNikita**
This PR provides `singlepass`  reducer for many usage scenarios

### CODE REVIEW

**Comment by michael-smirnov** [comment]
I think I would be better first to address comments in #791 and then split PR into the pieces - otherwise, comments can be lost/forgotten to apply.

**Comment by KulikovNikita** [comment]
@michael-smirnov First of all I would like to close successfully this PR. The #791 is marked as WIP and have the purpose to show current state of work on new kernels. I'll split it in the future. Thank you for valuable remark

File: cpp/oneapi/dal/backend/primitives/reduction/BUILD [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package(default_visibility = ["//visibility:public"])
+load("@onedal//dev/bazel:dal.bzl", "dal_module")
+
+dal_module(
+    name = "reducer",
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/reduction/BUILD:5]: Rename to match directory name -> reduction

File: cpp/oneapi/dal/backend/primitives/reduction/BUILD [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    dpc = True,
+    auto = True,
+    deps = [
+        "@onedal//cpp/oneapi/dal:common",
+        "@onedal//cpp/oneapi/dal/backend/primitives:common",
+    ],
+)
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/reduction/BUILD:12]: Add new line

File: cpp/oneapi/dal/backend/primitives/BUILD [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package(default_visibility = ["//visibility:public"])
+load("@onedal//dev/bazel:dal.bzl", "dal_module")
+
+dal_module(
+    name = "common",
+    auto = True,
+)
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/BUILD:7]: Add new line

File: cpp/oneapi/dal/backend/primitives/reduction/BUILD (hunk 1 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/reduction/BUILD]: Fixed

**Review by rlnx — COMMENTED** [review]
- I'd like to see simple interface for reduction as we were discussing.
- Better naming for `vector_size`, `n_vectors` aligned with the rest of the lib, like `row_count`, `column_count`.

**Comment by napetrov** [comment]
@KulikovNikita  - i would assume we don't need this PR?

**Pull request closed without merging**